internal/component/otelcol/internal/lazyconsumer/lazyconsumer.go

```diff
@@ -119,3 +139,15 @@ func (c *Consumer) SetConsumers(t otelconsumer.Traces, m otelconsumer.Metrics, l
 	c.logsConsumer = l
 	c.tracesConsumer = t
 }
+
+// Pause will stop the consumer until Resume is called. While paused, the calls to Consume* methods will block.
+// After calling Pause once, it must not be called again until Resume is called.
+func (c *Consumer) Pause() {
+	c.pauseMut.Lock()
+}
+
+// Resume will revert the Pause call and the consumer will continue to work. Resume must not be called if Pause wasn't
+// called before. See Pause for more details.
+func (c *Consumer) Resume() {
+	c.pauseMut.Unlock()
+}
```

Comment:
If pause is called by accident on a paused consumer or resume on a running consumer, it will be dramatic. Should the consumer holds its state (smth like "isPaused") to avoid locking if its already paused or avoid resuming if it's already running for extra safety?

Reply:
Yeah, I was thinking about this too, I wouldn't want to have any dramatic panics either. I've changed the implementation to handle multiple calls to Resume and Pause - it's a bit more complex, but I also wrote a lot of tests.